### PR TITLE
Aac spec infrastructure implementation

### PR DESCRIPTION
**Pull request by bunchw**
Implements project infrastructure from [AaC](https://github.com/DevOps-MBSE/AaC) into this repo.  Updates the spec definition.
Closes  [#784](https://github.com/DevOps-MBSE/AaC/issues/784) [#785](https://github.com/DevOps-MBSE/AaC/issues/785)

### CODE REVIEW

**Comment by lizzcondrey** [comment]
You seem to be missing the `README` for the `secure_install` directory. 

Need to update the package referencing to be the AaC  Specification package within the instructions. Including the file names of the `secure_install` directory.

The scripts referenced here and included in this directory are some of the pieces that might have to change with regards to the build steps of the pipeline that I mentioned might be additionally impacted by moving away from the `setup.py` build.

File: python/src/specifications/README.md
Comment (lizzcondrey) [inline python/src/specifications/README.md:1]: Think this will be what turns into the `aac-spec` page that I referenced in the `index.rst` page.

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

File: python/src/specifications/specifications.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+          type: str
+          description: The directory into which the CSV file will be generated.
+      acceptance:
+        - name: acceptance-criteria
+          scenarios: 
+            - name: acceptance-cases
+              given:
+                - The {{spec-csv.input.architecture-file}} contains a valid architecture specification.
+              when:
+                - The aac app is run with the spec-csv command.
```
Comment (lizzcondrey) [inline python/src/specifications/specifications.yaml:22]:
```suggestion
                - The aac-spec app is run with the spec-csv command.
```

File: .github/ISSUE_TEMPLATE/BUG-REPORT.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      description: Please answer the following questions for yourself before submitting an issue.
+      options:
+      - label: I checked the documentation and found no answer.
+        required: true
+      - label: I checked to make sure that this issue has not already been filed.
+        required: true
+  - type: input
+    id: version-pre-req
+    attributes:
+      label: "What version of the AaC package are you using? Check your version with the command: aac version."
```
[the remark this answers is not in the document]
Comment (bunchw) [reply .github/ISSUE_TEMPLATE/BUG-REPORT.yml:24]: Should I go ahead and include the spec version question?

File: python/src/specifications/specifications.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      helpText: Generate a CSV file listing requirements.
+      input:
+        - name: architecture-file
+          type: str
+          description:  The file to convert into CSV.
+        - name: output-directory
+          type: str
+          description: The directory into which the CSV file will be generated.
+      acceptance:
+        - name: acceptance-criteria
```
Comment (lizzcondrey) [inline python/src/specifications/specifications.yaml:16]:
```suggestion
        - name: Generate a Specification CSV
```

File: python/src/specifications/specifications.yaml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+          scenarios: 
+            - name: acceptance-cases
```
Comment (lizzcondrey) [inline python/src/specifications/specifications.yaml:18]:
```suggestion
            - name: Successful CSV Generation
```

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

File: .github/workflows/python-test-python-version.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python ${{ matrix.py_version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.py_version }}
+
+    - name: Install the AaC Package
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/python-test-python-version.yml:26]: @bunchw this still needs addressed.

File: .github/workflows/issues2md.yml
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/issues2md.yml:1]: Looks like this still needs to be removed.

File: .github/workflows/python-build-and-lint.yml [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+          cd ./docs
+          sphinx-build source doc_artifacts
+
+      - name: Upload Documentation
+        uses: actions/upload-artifact@v3
+        with:
+          name: aac-documentation
+          path: |
+            python/docs/doc_artifacts/api
+            python/docs/doc_artifacts/project_documentation
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/python-build-and-lint.yml:54]: @bunchw still needs addressed.

File: .github/workflows/python-cross-platform-support.yml [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python 3.9.13
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9.13
+
+      - name: Install the AaC Package
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/python-cross-platform-support.yml:26]: @bunchw still needs addressed

File: .github/workflows/python-deploy-artifacts.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
+name: Deploy Python Artifacts
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+    paths:
+      - python/**
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/python-deploy-artifacts.yml:9]: @bunchw still needs addressed

File: .github/workflows/python-test-python-version.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      run: pip install -e .[all]
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/python-test-python-version.yml]: @bunchw still needs addressed.

File: .github/workflows/sphinx.yml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Setup Pages
+        uses: actions/configure-pages@v3
+
+      - name: Build Documentation
+        uses: ammaraskar/sphinx-action@0.4
+        with:
+          pre-build-command: "pip install aac"
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .github/workflows/sphinx.yml:43]: @bunchw still needs addressed

File: .vscode/launch.json [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+            "justMyCode": true
+        },
+        {
+            "name": "AaC Gen-Plugin",
+            "type": "python",
+            "request": "launch",
+            "program": "${workspaceFolder}/python/src/aac/__main__.py",
+            "args": [
+                "gen-plugin",
+                "${workspaceFolder}/python/src/aac/plugins/no_ext_for_final/no_ext_for_final.aac",
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .vscode/launch.json:46]: @bunchw still needs addressed

File: .vscode/python.code-workspace [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+            "pydantic",
+            "pygls",
+            "pythonic",
+            "usecase",
+            "uvicorn"
+        ],
+        "python.languageServer": "Pylance",
+        "python.linting.mypyEnabled": true
+    },
+    "launch": {
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply .vscode/python.code-workspace:41]: @bunchw still needs addressed

File: docs/CNAME
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/CNAME:1]: Still needs removed.

File: docs/source/conf.py [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+import sys
+from datetime import date
+from importlib import metadata
+
+sys.path.insert(0, os.path.abspath("."))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "Architecture-as-Code-Specifications-Plugin"
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:23]: Still needs addressed.

File: docs/source/conf.py [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+copyright = f"2021 - {str(date.today().year)} , AaC Project Contributors"
+author = "AaC Project Contributors"
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:25]: Still needs addressed.

File: docs/source/conf.py [hunk 15]
```diff
@@ -0,0 +1,122 @@
...
+from datetime import date
+from importlib import metadata
+
+sys.path.insert(0, os.path.abspath("."))
+
+
+# -- Project information -----------------------------------------------------
+
+project = "Architecture-as-Code-Specifications-Plugin"
+copyright = f"2021 - {str(date.today().year)} , AaC Project Contributors"
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:24]: Still needs addressed.

File: docs/source/conf.py [hunk 16]
```diff
@@ -0,0 +1,122 @@
...
+author = "AaC Project Contributors"
+
+# The full version, including alpha/beta/rc tags
+release = metadata.version("aac")
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:28]: Still needs addressed.

File: python/requirements.in [hunk 17]
```diff
@@ -0,0 +1,15 @@
+build==1.1.1
+importlib-metadata==4.8
+pipdeptree==2.16.1
+sphinx==7.2.6
+myst-parser==2.0.0
+tox==4.14.1
+sphinx_contributors==0.2.7
+sphinx-copybutton==0.5.2
+sphinx-copybutton==0.5.2
```
[the remark this answers is not in the document]
Comment (bunchw) [reply python/requirements.in]: Removed one

File: python/requirements.in [hunk 18]
```diff
@@ -0,0 +1,15 @@
...
+furo==2024.1.29
+nose2==0.14.1
+coverage==7.4.3
+pyflakes == 2.4
+flake8 == 4.0
```
[the remark this answers is not in the document]
Comment (bunchw) [reply python/requirements.in]: Fixed

File: python/requirements.txt
[the remark this answers is not in the document]
Comment (bunchw) [reply python/requirements.txt:1]: Done

File: python/src/specifications/specifications.yaml (hunk 3 above)
Comment (bunchw) [reply python/src/specifications/specifications.yaml]: Changed

File: python/src/specifications/specifications.yaml (hunk 4 above)
Comment (bunchw) [reply python/src/specifications/specifications.yaml]: Changed

File: .github/workflows/python-cross-platform-support.yml [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+
+      - name: Set up Python 3.9.13
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9.13
+
+      - name: Install the AaC Specifications Package
+        run: python -m pip install -e
```
Comment (lizzcondrey) [inline .github/workflows/python-cross-platform-support.yml:27]:
```suggestion
        run: python -m pip install -e .
```
Still need the `.`, that says where to find what to install. The `[all]` was what portion of the `setup.py`  to be used with the install from that location. Was a tag within the `setup.py` script.

File: .github/workflows/python-test-python-version.yml [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+
+    - name: Set up Python ${{ matrix.py_version }}
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ matrix.py_version }}
+
+    - name: Install the AaC Specification Package
+      run: pip install -r requirements.txt
```
Comment (lizzcondrey) [inline .github/workflows/python-test-python-version.yml:27]:
```suggestion
      run: pip install -e .
```

I don't think swapping to an install of just the requirements gets what was desired of a package install.

File: docs/source/conf.py (hunk 15 above)
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:24]: @bunchw missed one.

File: docs/source/conf.py (hunk 16 above)
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/conf.py:28]: @bunchw missed one

File: docs/source/index.rst [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+
+   api
+
+
+Indices and tables
+------------------
+
+* :ref:`genindex`
+* :ref:`modindex`
+* :ref:`search`
```
[the remark this answers is not in the document]
Comment (lizzcondrey) [reply docs/source/index.rst:21]: @bunchw missed

**Review by lizzcondrey — CHANGES_REQUESTED** [review]

File: .github/workflows/python-cross-platform-support.yml (hunk 19 above)
Comment (bunchw) [reply .github/workflows/python-cross-platform-support.yml]: Whoops, fixed.

File: docs/source/index.rst [hunk 22]
```diff
@@ -0,0 +1,4 @@
+The AaC Specification (aac-spec) Plugin is an extension of the core AaC tool and provides specification and requirements traceability capabilities upon ingest of an AaC model. 
+Documentation for the core AaC tool can be found `here <https://arch-as-code.org/>`_"
```
Comment (lizzcondrey) [inline docs/source/index.rst:2]:
```suggestion
Documentation for the core AaC tool can be found `here <https://arch-as-code.org/>`_
```

Missed removing an end quote from copying in the comment in the initial suggestion.

**Review by lizzcondrey — CHANGES_REQUESTED** [review]
Last one I think. Great job getting things addressed.

File: docs/source/index.rst (hunk 22 above)
Comment (bunchw) [reply docs/source/index.rst]: Done

**Review by lizzcondrey — APPROVED** [review]
That should get everything to a solid and safe start point for the new repo. Good job.

**Pull request merged**